### PR TITLE
gateway: sending sms messages (+mock tests)

### PR DESCRIPTION
**Pull request by martinmilev**
Ready for review.

### CODE REVIEW

File: src/com/clouway/sms/GateWay.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+public class GateWay {
+
+  public boolean send(Sms sms, Validator v){
+    if(v.validate(sms)){
+      return true;
+    }
+    return false;
+  }
+
+  public boolean recieved(Sms sms, Receiver r) throws Exception {
```
Comment (mgenov) [inline src/com/clouway/sms/GateWay.java]: This method should not be part of the Gateway. 

Gateway#send calls Validator to validate message 
Geteway#send calls Receiver to deliver the message 
 
t.e Gateway should have only one method.

File: test/com/clouway/sms/SmsValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * @Author Martin Milev
+ * @Date 07.06.16.
+ */
+public class SmsValidationTest {
```
Comment (mgenov) [inline test/com/clouway/sms/SmsValidationTest.java]: You are testing the Gateway here, not the validation.

File: test/com/clouway/sms/SmsReceiverTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * @Author Martin Milev
+ * @Date 08.06.16.
+ */
+public class SmsReceiverTest {
```
Comment (mgenov) [inline test/com/clouway/sms/SmsReceiverTest.java]: You are testing the Gateway here.... not the receiver.

Comment (mgenov) [inline test/com/clouway/sms/SmsReceiverTest.java]: Actually both tests need to be combined when single "send" method is used.

File: test/com/clouway/sms/SmsValidatorTest.java [hunk 4]
```diff
@@ -12,66 +12,59 @@
  * @Date 07.06.16.
  */
 public class SmsValidatorTest {
```
Comment (mgenov) [inline test/com/clouway/sms/SmsValidatorTest.java]: Name of the class is misleading.

File: test/com/clouway/sms/SmsValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+
+/**
+ * @Author Martin Milev
+ * @Date 07.06.16.
+ */
+public class SmsValidatorTest {
```
Comment (mgenov) [inline test/com/clouway/sms/SmsValidatorTest.java]: You don't need Gateway or Forwarder for testing the SmsValidator.

**Comment by mgenov** [comment]
Any progress on comments?

File: test/com/clouway/sms/SmsValidatorTest.java (hunk 5 above)
Comment (martinmilev) [inline test/com/clouway/sms/SmsValidatorTest.java]: Testing only the SmsValidator, now.

File: test/com/clouway/sms/SmsValidatorTest.java (hunk 4 above)
Comment (martinmilev) [inline test/com/clouway/sms/SmsValidatorTest.java]: Changed the name to Validation test. Don't know if that is acceptable.

**Comment by martinmilev** [comment]
I'v dome some changes on the code. Don't know if writing on closed comments sends notifications.

**Comment by mgenov** [comment]
LGTM !

**Comment by mgenov** [comment]
Yeah comments are received, but you commented only to 2 of them.

**Pull request merged**